### PR TITLE
lint: require "await" or .catch() on async calls

### PR DESCRIPTION
**Pull request by justinmk3**
Continued from:

- https://github.com/aws/aws-toolkit-vscode/pull/4202
- #1638

# Problem:

- All promises or async calls should use `await`, or end with `.catch()`.
  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises
- Async calls accidentally forget `await`.
  - Example: 8863ba4a799e
- Test logs show "rejected promise not handled":
  ```
  rejected promise not handled within 1 second: HTTPError: Response code 403 (rate limit exceeded)
  rejected promise not handled within 1 second: Error: command 'aws.codeWhisperer.refreshStatusBar' not found
  rejected promise not handled within 1 second: HTTPError: Response code 404 (Not Found)
  rejected promise not handled within 1 second: CodeExpectedError: cannot open notCloudWatch:hahahaha. Detail: Unable to resolve resource notCloudWatch:hahahaha
  rejected promise not handled within 1 second: AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  rejected promise not handled within 1 second: Error: No access
  rejected promise not handled within 1 second: Error: No access
  rejected promise not handled within 1 second: Error: Timed-out waiting for browser login flow to complete
  rejected promise not handled within 1 second: Error: Download code bindings cancelled
  rejected promise not handled within 1 second: Error: EACCES: permission denied, mkdir '/my'
  ```

# Solution:

- Enable eslint rule: https://typescript-eslint.io/rules/no-floating-promises/

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/codewhisperer/tracker/codewhispererTracker.test.ts [hunk 1]
```diff
@@ -65,9 +65,11 @@ describe('codewhispererTracker', function () {
         })
 
         it('Should skip if telemetry is disabled', async function () {
+            globals.telemetry.telemetryEnabled = false
             const getTimeSpy = sinon.spy(Date.prototype, 'getTime')
             await CodeWhispererTracker.getTracker().flush()
             assert.ok(!getTimeSpy.called)
```
Comment (justinmk3) [inline src/test/codewhisperer/tracker/codewhispererTracker.test.ts:71]: Before the change to `codewhispererTracker.ts`, this test was accidentally passing.

**Comment by andrewyuq** [comment]
So in case when we don't need to await we will just use `void`?

**Comment by justinmk3** [comment]
> So in case when we don't need to await we will just use `void`?

You  always should `await` or `.catch` except when it's not possible.

For this initial effort, I've used `void` in many places simply because manually deciding how to handle 500 different call sites is an undue burden. But that doesn't mean using `void` is an acceptable practice.

**Pull request merged**